fedimint-core/src/encoding/mod.rs

```diff
@@ -248,17 +249,103 @@ macro_rules! impl_encode_decode_tuple {
     );
 }
 
+/// Specialized version of Encodable for bytes
+pub fn consensus_encode_bytes<W: std::io::Write>(
+    bytes: &[u8],
+    writer: &mut W,
+) -> Result<usize, Error> {
+    let mut len = 0;
+    len += (bytes.len() as u64).consensus_encode(writer)?;
+    writer.write_all(bytes)?;
+    len += bytes.len();
+    Ok(len)
+}
+
+/// Specialized version of Encodable for static byte arrays
+pub fn consensus_encode_bytes_static<const N: usize, W: std::io::Write>(
+    bytes: &[u8; N],
+    writer: &mut W,
+) -> Result<usize, Error> {
+    writer.write_all(bytes)?;
+    Ok(bytes.len())
+}
+
+struct ReadBytesFromFiniteReaderOpts {
+    len: usize,
+    chunk_size: usize,
+}
+
+/// Read `opts.len` bytes from reader, where `opts.len` could potentially be
```

Comment:
This reminds me I need to port https://github.com/rust-bitcoin/rust-bitcoin/blob/e2b9555070d9357fb552e56085fb6fb3f0274560/bitcoin/src/consensus/encode.rs#L304 to Fedimint ...